src/components/SearchField.tsx

```diff
@@ -1,38 +1,241 @@
-import React, { useState } from 'react';
+import React, { useCallback, useState } from 'react';
+import { useHistory } from 'react-router';
+
+import '../styles/SearchField.scss';
 
 const loupeIcon = '/images/search-icon.svg';
 
+const genres = [
+  'none',
+  'Comedy',
+  'Drama',
+  'Romance',
+  'Mystery',
+  'Thriller',
+  'Animation',
+  'Action',
+  'Adventure',
+  'Crime',
+  'Fantasy',
+  'Science Fiction',
+  'Horror',
+  'Family'
+];
+
+const checkedImg = '/images/searchFieldIcons/checked.svg';
+const uncheckedImg = '/images/searchFieldIcons/unchecked.svg';
+
 const SearchField: React.FC = () => {
+  const history = useHistory();
+
   const [inputValue, setInputValue] = useState<string>('');
 
-  const changeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+  const [focus, setFocus] = useState<boolean>(false);
+
+  const [dropdown, setDropdown] = useState<boolean>(false);
+
+  const [dateRange, setDateRange] = useState<{
+    fromDate: number;
+    toDate: number;
+  }>({
```

Comment:
probably would be easier just to use separate states for dateFrom/dateTo

Reply:
Now it works properly, may I leave it for now?